core/include/mbox.h

```diff
@@ -185,6 +185,17 @@ static inline size_t mbox_avail(mbox_t *mbox)
     return cib_avail(&mbox->cib);
 }
 
+/**
+ * @brief   Unset's the mbox, effectively deinitializing invalidating it.
```

Comment:
both deinitializing and invalidating? Comma missing? :)

This leaks a possibly supplied queue. Maybe better call it mbox_release_queue?
The mbox actually works without queue, it'd just be synchronous from that point on.

This is also a bit dangerous, if this is called with messages in the queue, they get lost. If there's a sender that was previously waiting, it would be stuck there until the queue transitions from full to empty again.

Why is this needed?

Reply:
(I suggest splitting, the other changes were already ACKed...)

Reply:
> Why is this needed?

Scroll the code and you see, that lwIP requires a way to invalidate their mboxes and check the validiy.

https://github.com/RIOT-OS/RIOT/blob/dd1de91c636bc98956bc40fd904df2ee4b213ed2/pkg/lwip/include/arch/sys_arch.h#L101-L111

I agree that the current approach is not the correct one, maybe. Any idea how we should proceed with that?


> (I suggest splitting, the other changes were already ACKed...)

Can do.

Reply:
See https://github.com/RIOT-OS/RIOT/pull/15485